crates/uv/src/commands/python/install.rs

```diff
@@ -405,14 +437,44 @@ pub(crate) async fn install(
         }
     }
 
+    // Read all existing installations and find the highest installed patch
+    // for each installed minor version.
+    let installations = ManagedPythonInstallations::from_settings(None)?.init()?;
+    let existing_installations: Vec<_> = installations.find_all()?.collect();
+    let mut minor_versions = FxHashMap::default();
+    for installation in existing_installations {
+        // Add to minor versions map if this installation has the highest
+        // patch seen for a minor version so far.
+        let minor_version = installation.version().python_version();
+        minor_versions
+            .entry(minor_version)
+            .and_modify(|high_installation: &mut ManagedPythonInstallation| {
+                if installation.key() >= high_installation.key() {
+                    *high_installation = installation.clone();
+                }
+            })
+            .or_insert(installation);
+    }
+
+    for installation in minor_versions.values() {
+        installation.ensure_minor_version_link()?;
+    }
+
     if changelog.installed.is_empty() && errors.is_empty() {
         if is_default_install {
             writeln!(
                 printer.stderr(),
                 "Python is already installed. Use `uv python install <request>` to install another version.",
             )?;
         } else if requests.len() > 1 {
-            writeln!(printer.stderr(), "All requested versions already installed")?;
+            if upgrade {
+                writeln!(
+                    printer.stderr(),
+                    "All requested versions already on latest patch"
```

Comment:
```suggestion
                    "All requested versions already on latest supported patch release"
```

This may be confusing otherwise due to uv bundling supported patch versions.